testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -161,6 +161,31 @@ def test_AnalysisFromFunction(u, start, stop, step, nframes):
     for ana in (ana1, ana2, ana3):
         assert_equal(results, ana.results)
 
+        
+def mass_xyz(atomgroup, masses):
+        return atomgroup.positions * masses
+
+def test_AnalysisFromFunction_args_length(u):
+    protein = u.select_atoms('protein')
+    masses = protein.masses.reshape(-1, 1)
+    ans = base.AnalysisFromFunction(mass_xyz, protein, masses)
+    assert len(ans.args) == 2
+    assert int(sum(sum(sum(ans.run().results)))) == -317054
+
+def test_AnalysisFromFunction_args_content(u):
+    protein = u.select_atoms('protein')
+    masses = protein.masses.reshape(-1, 1)
+    ans = base.AnalysisFromFunction(mass_xyz, protein, masses)
+    assert (ans.args[0] == protein) and not (ans.args[1] - masses).any()
+    assert  ans._trajectory == protein.universe.trajectory
+
+def test_AnalysisFromFunction_args_multiple_traj(u):
+    protein = u.select_atoms('protein')
+    masses = protein.masses.reshape(-1, 1)
+    another = mda.Universe(TPR, XTC).select_atoms("protein")
+    ans = base.AnalysisFromFunction(mass_xyz, another, protein, masses)
+    assert ans._trajectory == another.universe.trajectory
```

Comment:
@lilyminium is this what it should be testing here?

Reply:
Yes, although if you use [is instead of ==](http://net-informations.com/python/iq/is.htm) you could avoid importing new files. In general, using `is` would simplify these tests.

Also, `mass_xyz` wouldn't actually run here.

Reply:
okay, I will change `==` to `is`. Even `mass_xyz` couldn't run here, we can still test the args order, is this a problem?

Reply:
@lilyminium I have revised `mass_xyz` which can run given two trajs.